Fix shapes of ocular frames

10110111 opened the pull request:
### Description

Now sensor frame, crop overlay, crop overlay grid, OAG frame its inner and outer circles, and the focuser circles — all are computed in terms of view direction, and correctly projected according to the current projection. This results in non-rectangular/non-circular shapes in projections other than perspective, which is to be expected, and is a good thing.

The problem remains with positioning of the text, and here I'd like to hear some opinions on how it should be fixed: should I try to rotate it along the tangent to the curve that represents the frame (see the last screenshot), or maybe better to just move it closer to the curve?

Fixes #3249 

### Screenshots

#### Perspective projection, old

![stellarium-010](https://github.com/Stellarium/stellarium/assets/6376882/706b20fc-9093-428e-98ed-84e462912fb6)

#### Perspective projection, new

![stellarium-011](https://github.com/Stellarium/stellarium/assets/6376882/91175ed6-c71e-4dc1-b4aa-8bdebf553a3d)

#### Stereographic projection, old

![stellarium-008](https://github.com/Stellarium/stellarium/assets/6376882/c8b79cdf-5645-48c2-9d97-adfc319f261f)

#### Stereographic projection, new

![stellarium-009](https://github.com/Stellarium/stellarium/assets/6376882/5a0a81ba-c10f-45e5-997d-637d37537caa)

### Type of change
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

**Test Configuration**:
* Operating system: Ubuntu 20.04
* Graphics Card: Intel UHD Graphics 620

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

gzotti commented:
Nice work!

The text placement should be visually guided by the frame corners. The easiest in the stereographic example may be to use a boundingrect for the pincushion shape and show text along its frame. Not sure how a tangential but straight orientation would look.  Of course, following the arc of the frame would be extremely elegant. But I am not sure how weird a frame rectangle may look in cylindrical or Hammer-Aitoff projections, so this may become complicated.

10110111 commented:
> did you comparing new frame and photo from the report?

Yes, and at least the frame now has correct shape. The scale is not ideal, but this may be due to unspecified way lens distortions are corrected by the camera, so that we don't know the exact effective focal length. At least theoretically, the frame rendering now works correctly, I checked this using Cursor Coordinates plugin to check the fields of view.

10110111 commented:
OK, I've now fixed the positioning of the text around the sensor frame. Note that "Max exposure" label is now moved to top-right position, because top row has two lines, with both being occupied on the left side, but only one on the right side. This way the overall look becomes a bit more symmetric.

Screenshots:

![stellarium-010](https://github.com/Stellarium/stellarium/assets/6376882/24cd38e0-030a-4a2d-8b10-ba5fce0ff598)

![stellarium-011](https://github.com/Stellarium/stellarium/assets/6376882/f45e8eb8-8850-4346-b800-f895c92fca3b)

alex-w reviewed (approved):
Thanks!

gzotti commented:
If the telescope (photo lens) has equatorial mount set, the labels are now off. 
![stellarium-083](https://github.com/Stellarium/stellarium/assets/18099873/9ed4217c-9c82-4faf-8785-07e3cf24c424)

10110111 commented:
Should be fixed now.
